### PR TITLE
Delete button for filters

### PR DESCRIPTION
**Pull request by frederike-ramin**
1) Rename Inputfield zo RemovableIputField
2) Add delete button to filters on detail page (same behavior as deleting the input field's value)

### CODE REVIEW

File: src/Utils/RemovableInputField.js [hunk 1]
```diff
@@ -1,16 +1,29 @@
...
         //Function binding
         this.handleInput = this.handleInput.bind(this);
+        this.handleRemove = this.handleRemove.bind(this);
     }
 
     handleInput(event) {
         this.props.onInputChange(this.props.id, event.target.value);
     }
 
+    handleRemove(event) {
```
Comment (hpibptlab) [inline src/Utils/RemovableInputField.js]: MAJOR Remove the unused function parameter "event". [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

File: src/ProductView/FilterBar/FilterBar.js [hunk 2]
```diff
@@ -1,14 +1,23 @@
 import React, { Component } from 'react';
-import InputField from '../../Utils/InputField/InputField.js';
+import RemovableInputField from '../../Utils/RemovableInputField.js';
+
 
 class FilterBar extends Component {
+    decideOnRemoveButton(currentFilter) {
+        if(this.props.filter[this.props.filter.length - 1] === currentFilter) {
```
Comment (hpibptlab) [inline src/ProductView/FilterBar/FilterBar.js]: MINOR Replace this if-then-else statement by a single return statement. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1126)

**Comment by hpibptlab** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: src/App/App.scss [hunk 3]
```diff
@@ -46,6 +46,13 @@ h4 {
     border-style: none none solid none;
     border-radius: 0;
   }
+  #remove-addon {
+    background-color: transparent;
+    color: $COLOR-dark-background;
+    border-color: $COLOR-dark-background;
+    border-style: none none solid none;
```
Comment (friedow) [inline src/App/App.scss:53]: If I click the delete button, there is a border / shadow appearing for a short time. This should not happen, therefore please check the button for :hover / :focus / :active css attributes.

File: src/Utils/RemovableInputField.js [hunk 4]
```diff
@@ -1,16 +1,29 @@
 import React, { Component } from 'react';
 
-class InputField extends Component {
+class RemovableInputField extends Component {
```
Comment (friedow) [inline src/Utils/RemovableInputField.js:3]: Why have you renamed the component?

File: src/ProductView/FilterBar/FilterBar.js [hunk 5]
```diff
@@ -1,14 +1,23 @@
...
+
     render() {
         return (
             <div className="filter-bar container">
                 <div className="form-group row">
                     {this.props.filter.map((filter) =>
-                        <InputField key={filter.id} id={filter.id} value={filter.value} placeholder="Search..."
-                                    onInputChange={this.props.onInputChange}/>
+                        <RemovableInputField key={filter.id} id={filter.id} value={filter.value} placeholder="Search..."
+                                             onInputChange={this.props.onInputChange} showRemove={this.decideOnRemoveButton(filter)}/>
```
Comment (friedow) [inline src/ProductView/FilterBar/FilterBar.js]: Please rename "showRemove" to "displayRemoveButton" for a better understanding.

File: src/ProductView/FilterBar/FilterBar.js [hunk 6]
```diff
@@ -1,14 +1,23 @@
...
+    }
+
     render() {
         return (
             <div className="filter-bar container">
                 <div className="form-group row">
                     {this.props.filter.map((filter) =>
-                        <InputField key={filter.id} id={filter.id} value={filter.value} placeholder="Search..."
-                                    onInputChange={this.props.onInputChange}/>
+                        <RemovableInputField key={filter.id} id={filter.id} value={filter.value} placeholder="Search..."
```
Comment (friedow) [inline src/ProductView/FilterBar/FilterBar.js]: Please break this component initialisation after each parameter.
Corresponding convention: If a component initialisation has more than 2 parameters, insert a line break after each parameter.
(I'll add the development conventions to the readme.)

File: src/Utils/RemovableInputField.js [hunk 7]
```diff
@@ -1,16 +1,29 @@
...
         this.props.onInputChange(this.props.id, event.target.value);
     }
 
+    handleRemove(event) {
+        this.props.onInputChange(this.props.id, '');
+    }
+
+    addRemoveButton() {
+        if (this.props.showRemove){
+            return <button className="input-group-addon" id="remove-addon" onClick={this.handleRemove}>
```
Comment (friedow) [inline src/Utils/RemovableInputField.js]: Please correct the indentation and wrap the jsx in round brackets. (For a template just look on the render method.)

**Review by friedow — CHANGES_REQUESTED** [review]

File: src/Utils/RemovableInputField.js (hunk 4 above)
Comment (frederike-ramin) [reply src/Utils/RemovableInputField.js:3]: It is a lot more descriptive name. Not every input field should be removable. that's the speciality about this input field. That's why a new name.

File: src/App/App.scss (hunk 3 above)
Comment (frederike-ramin) [reply src/App/App.scss:53]: Does the new :outline attribute fix that for you?

Comment (friedow) [reply src/App/App.scss:53]: Yep you fixed it, thanks.

File: src/Utils/RemovableInputField.js (hunk 4 above)
Comment (friedow) [reply src/Utils/RemovableInputField.js:3]: You're right.

**Review by friedow — APPROVED** [review]

**Pull request merged**